spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,429 @@
+          .typeParameterClass(Long.class)
+          .build();
+
+  public static final ConfigOption<Long> SECONDARY_RESOURCE_CREATE_MAX_ATTEMPTS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.retry.secondaryResourceCreateMaxAttempts")
+          .description(
+              "Maximal number of retry attempts of requesting secondary resource for Spark "
+                  + "application. This would be performed on top of k8s client "
+                  + "spark.kubernetes.operator.retry.maxAttempts to overcome potential "
+                  + "conflicting reconcile on the same SparkApplication. ")
+          .defaultValue(3L)
+          .typeParameterClass(Long.class)
+          .build();
+
+  public static final ConfigOption<Boolean> JOSDK_METRICS_ENABLED =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.metrics.josdkMetricsEnabled")
+          .description(
+              "When enabled, the josdk metrics will be added in metrics source and "
+                  + "configured for operator.")
+          .defaultValue(true)
+          .build();
+
+  public static final ConfigOption<Boolean> KUBERNETES_CLIENT_METRICS_ENABLED =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.metrics.clientMetricsEnabled")
+          .defaultValue(true)
+          .description(
+              "Enable KubernetesClient metrics for measuring the HTTP traffic to "
+                  + "the Kubernetes API Server. Since the metrics is collected "
+                  + "via Okhttp interceptors, can be disabled when opt in "
+                  + "customized interceptors.")
+          .build();
+
+  public static final ConfigOption<Boolean>
+      KUBERNETES_CLIENT_METRICS_GROUP_BY_RESPONSE_CODE_GROUP_ENABLED =
+          ConfigOption.<Boolean>builder()
+              .key("spark.kubernetes.operator.metrics.clientMetricsGroupByResponseCodeEnabled")
+              .description(
+                  "When enabled, additional metrics group by http response code group(1xx, "
+                      + "2xx, 3xx, 4xx, 5xx) received from API server will be added. Users "
+                      + "can disable it when their monitoring system can combine lower level "
+                      + "kubernetes.client.http.response.<3-digit-response-code> metrics.")
+              .defaultValue(true)
+              .build();
+
+  public static final ConfigOption<Integer> OPERATOR_METRICS_PORT =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.metrics.port")
+          .defaultValue(19090)
+          .description("The port used for checking metrics")
+          .typeParameterClass(Integer.class)
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<Integer> OPERATOR_PROBE_PORT =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.health.probePort")
+          .defaultValue(18080)
```

Comment:
Please choose a different value because this port is used as a well-known Apache Spark History Server port? If you can propose a unique port number it would be great for the whole Spark community because nobody will get confused.

Reply:
+1, moved to 19091